### PR TITLE
Namespace external metrics

### LINKED ISSUE #255: external metrics are not namespace scoped for prometheus
## Expected Behavior
Assuming you have two different deployment of the same application running in the same cluster but in different namespaces. You define the same query name like this. hpa 1 and 2 use the same name. The expected result for me is that the hpas in different namespaces are not ware of each other and cant interfere.

hpa 1
```yaml
    metric-config.external.fr-hpa-http-requests.prometheus/query: |
      scalar(sum(rate(fr_hpa_testing_http_requests_total{kubernetes_namespace="alpha"}[5m])))
``` 

hpa 2
```yaml
    metric-config.external.fr-hpa-http-requests.prometheus/query: |
      scalar(sum(rate(fr_hpa_testing_http_requests_total{kubernetes_namespace="beta"}[5m])))
``` 

## Actual Behavior
Since they use the same name and you see one registered external metric they interfere with each other and the HPA scales on the same metric not differently 

example output showing this 
```bash
fr-hpa-testing main ? ❯❯❯ k get hpa -A 
NAMESPACE   NAME                 REFERENCE                   TARGETS           MINPODS   MAXPODS   REPLICAS   AGE
alpha       fr-hpa-testing-hpa   Deployment/fr-hpa-testing   14440m/50 (avg)   1         3         1          37m
beta        fr-hpa-testing-hpa   Deployment/fr-hpa-testing   14440m/50 (avg)   1         3         1          37m
fr-hpa-testing main ? ❯❯❯ kubectl get -A  --raw "/apis/external.metrics.k8s.io/v1beta1" | jq .
{
  "kind": "APIResourceList",
  "apiVersion": "v1",
  "groupVersion": "external.metrics.k8s.io/v1beta1",
  "resources": [
    {
      "name": "fr-hpa-http-requests",
      "singularName": "",
      "namespaced": true,
      "kind": "ExternalMetricValueList",
      "verbs": [
        "get"
      ]
    }
  ]
}

```

## Specifications

  - Version: 2bfc85ff3d48039d8f36af8d8aee64b204e99048
  - Platform: k8s 1.17
  - Subsystem:

### PR DESCRIPTION
**Pull request by mikkeloscar**
This introduces namespacing of external metrics such that metrics from two different namespaces with the same configuration (name/labels) don't overlap.

This is done by:
1. Changing the metric store to index by namespace
2. Adding namespace when setting up a collector and pass namespace via the CollectedMetric.

Fix #255

### CODE REVIEW

**Comment by mikkeloscar** [comment]
:+1:

**Comment by mikkeloscar** [comment]
:+1:

File: pkg/collector/prometheus_collector.go [hunk 1]
```diff
@@ -135,7 +135,7 @@ func NewPrometheusCollector(client kubernetes.Interface, promAPI promv1.API, hpa
 
 func (c *PrometheusCollector) GetMetrics() ([]CollectedMetric, error) {
 	// TODO: use real context
-	value, _, err := c.promAPI.Query(context.Background(), c.query, time.Now().UTC())
+	value, _, err := c.promAPI.Query(context.TODO(), c.query, time.Now().UTC())
```
Comment (szuecs) [inline pkg/collector/prometheus_collector.go:138]: this change seems unrelated

Comment (mikkeloscar) [reply pkg/collector/prometheus_collector.go]: dropped it from this PR.

**Comment by mikkeloscar** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:
